plugins/c9.vfs.client/vfs_client.js

```diff
@@ -193,12 +193,12 @@ define(function(require, exports, module) {
                 extraPaths = path;
                 path = path[0];
                 extraPaths = "," + extraPaths.map(function(p) {
-                    return p[0] == path[0] && p != path ? escape(p) : "";
+                    return p[0] == path[0] && p != path ? encodeURI(p) : "";
```

Comment:
You are right, touching the escape in vfsUrl is risky.
But with this server calls unescape on result of encodeURI which is not correct.
Try running the following in the browser

```js
[
, original = "/as/f+%2C/аաe"
, encoded = encodeURI(original)
, encodedAndUnescaped = unescape(encoded)
, encodedAndUnescaped == original
, escaped = escape(original)
, escapedAndUnescaped = unescape(escaped)
, original == escapedAndUnescaped
]
```
Since error was only in filename, then

```diff
-                + (filename ? "=" + escape(filename) : "")
+                + (filename ? "=" + encodeURIComponent(filename) : "")
```

should be enough right?